<?xml version="1.0"?>

<launch>

    <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 map odom 30" />
    <node pkg="sar_SLAM" type="odom_tf" name="tf2_odom_broadcaster"/>
    <node pkg="sar_SLAM" type="lidar2odom" name="lidar2odom">
      <remap from="odom_lidar" to="/odom"/>
    </node>

    

    <arg name="tf_map_scanmatch_transform_frame_name" default="scanmatcher_frame"/>
    <arg name="base_frame" default="base_footprint"/>
    <arg name="odom_frame" default="base_footprint"/>
    <!--<arg name="odom_frame" default="base_footprint"/>-->
    <arg name="pub_map_odom_transform" default="true"/>
    <arg name="scan_subscriber_queue_size" default="5"/>
    <arg name="scan_topic" default="/scan"/>
    <arg name="map_size" default="2048"/>
    
    <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
      
      <!-- Frame names -->
      <param name="map_frame" value="map" />
      <param name="base_frame" value="$(arg base_frame)" />
      <param name="odom_frame" value="$(arg odom_frame)" />
      
      <!-- Tf use -->
      <param name="use_tf_scan_transformation" value="true"/>
      <param name="use_tf_pose_start_estimate" value="false"/>
      <param name="pub_map_odom_transform" value="$(arg pub_map_odom_transform)"/>
     
      
      <!-- Map update parameters -->
      <param name="update_factor_free" value="0.4"/>
      <param name="update_factor_occupied" value="0.9" />    
      <param name="map_update_distance_thresh" value="0.4"/>
      <param name="map_update_angle_thresh" value="0.06" />
      <param name="laser_z_min_value" value = "-1.0" />
      <param name="laser_z_max_value" value = "1.0" />
      
      <!-- Advertising config --> 
      <param name="advertise_map_service" value="true"/>
      
      <param name="scan_subscriber_queue_size" value="$(arg scan_subscriber_queue_size)"/>
      <param name="scan_topic" value="$(arg scan_topic)"/>
      
      <!-- Debug parameters -->
      <!--
        <param name="output_timing" value="false"/>
        <param name="pub_drawings" value="true"/>
        <param name="pub_debug_output" value="true"/>
      -->
      <param name="tf_map_scanmatch_transform_frame_name" value="$(arg tf_map_scanmatch_transform_frame_name)" />
    </node>
      
    <!--<node pkg="tf" type="static_transform_publisher" name="map_nav_broadcaster" args="0 0 0 0 0 0 map nav 100"/>-->
    <node pkg="rviz" type="rviz" name="rviz" required="true"
    args="-d $(find sar_SLAM)/rviz/hector_slam.rviz"/>
  </launch>
